cmd/kube-proxy/app/server.go

```diff
@@ -745,11 +743,7 @@ func (s *ProxyServer) Run() error {
 	serviceConfig.RegisterEventHandler(s.Proxier)
 	go serviceConfig.Run(wait.NeverStop)
 
-	if endpointsHandler, ok := s.Proxier.(config.EndpointsHandler); ok && !s.UseEndpointSlices {
-		endpointsConfig := config.NewEndpointsConfig(informerFactory.Core().V1().Endpoints(), s.ConfigSyncPeriod)
-		endpointsConfig.RegisterEventHandler(endpointsHandler)
-		go endpointsConfig.Run(wait.NeverStop)
-	} else {
+	if _, ok := s.Proxier.(config.EndpointsHandler); ok {
```

Comment:
I think the EndpointsHandler is not used in the RegisterEventHandler, meaning all the proxy/config related to the endpoint can also be removed - this check can be cleaned.